bin/reth/src/node/mod.rs

```diff
@@ -271,27 +271,22 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
         let blockchain_db = BlockchainProvider::new(factory, blockchain_tree.clone())?;
         let blob_store = InMemoryBlobStore::default();
 
-        let validator = if let Some(ref trusted_setup_file) = self.trusted_setup_file {
-            let kzg_settings = self.load_kzg_settings(trusted_setup_file)?;
-            info!(target: "reth::cli", "KZG trusted setup file loaded");
-
-            TransactionValidationTaskExecutor::eth_with_kzg_settings_and_tasks(
-                blockchain_db.clone(),
-                Arc::clone(&self.chain),
-                blob_store.clone(),
-                ctx.task_executor.clone(),
-                1,
-                kzg_settings,
-            )
-        } else {
-            TransactionValidationTaskExecutor::eth_with_additional_tasks(
-                blockchain_db.clone(),
-                Arc::clone(&self.chain),
-                blob_store.clone(),
-                ctx.task_executor.clone(),
-                1,
-            )
+        let eth_builder = TransactionValidationTaskExecutor::eth_builder(Arc::clone(&self.chain));
+        // Overwrites the trusted setup file.
+        let eth_builder = match self.trusted_setup_file {
+            Some(ref trusted_setup_file) => {
+                let kzg_settings = self.load_kzg_settings(trusted_setup_file)?;
+                info!(target: "reth::cli", "KZG trusted setup file loaded");
+
+                eth_builder.kzg_settings(kzg_settings)
+            }
+            None => eth_builder,
```

Comment:
nit: I'd like a function that returns `Arc<KzgSettings>` here, so we can do

```
TransactionValidationTaskExecutor::eth_builder(Arc::clone(&self.chain)).kzg_settings(self.kzg_settings()?)
```

Reply:
I might need some help here. Since `self.trusted_setup_file` is an `Option<PathBuf>`, we can't set the `kzg_settings` when the user doesn't pass the trusted setup file to the cli, because we need to handle the `None` case.

Personally, I wouldn't like to change `fn kzg_settings(mut self, kzg_settings: Arc<KzgSettings>)` to accept an `Option`. Lmk your thoughts.

Reply:
Ah, gotcha, makes sense 👍🏻